lib/IAMClient.js

```diff
@@ -979,6 +979,40 @@ class VaultClient {
         }, data, options.reqUid, null, options.logger);
     }
 
+    /**
+     * Retrieves the default encryption key id for the given account's canonical id,
+     * or creates one if it does not exist.
+     *
+     * @param {String} canonicalId - The canonical id of the account.
+     * @param {Object} options - Additional arguments.
+     * @param {string} options.reqUid - The request UID.
+     * @param {Function} callback
+     * - `error` (Error|null) - error object if the operation failed, otherwise null.
+     * - `result` (Object) - result object on success, containing raw response data and HTTP status code.
+     * @returns {void}
+     */
+    getOrCreateEncryptionKeyId(canonicalId, options, callback) {
+        assert(canonicalId !== undefined, 'canonicalId need to be specified');
+        assert(typeof canonicalId === 'string', 'canonicalId should be a string');
+        assert(canonicalId !== '', 'canonicalId should not be empty string');
+        const data = {
+            Action: 'GetOrCreateEncryptionKeyId',
+            canonicalId,
+        };
+        this.request('POST', '/', false, (err, data, code) => {
```

Comment:
Is a POST to / doing what we want to do here? 🤔

Reply:
Yes, :) It uses an RPC (remote procedure call) protocol. All actions are sent to a single endpoint `'/'` and the payload specifies the action to do:
```
const data = {
      Action: 'GetOrCreateEncryptionKeyId',
      canonicalId,
  };
```

Reply:
Ooh I see. I'm not familiar with the vault API and I assumed it uses HTTP routes.

Reply:
This action doesn't exist for now I think